src/js/controllers/mapController.ts

```diff
@@ -1433,96 +1437,102 @@ export class MapController {
+      const modisIds = MODISLayerIDs.map(l => l.id);
+      this._map!.layers.forEach(l => {
+        if (modisIds.includes(l.id)) {
+          l.visible = l.id === 'MODIS1Y';
         }
-      );
-
-      store.dispatch(allAvailableLayers(newLayersArray));
+      });
+      store.dispatch(setModisStart(String(startDate)));
+      store.dispatch(setModisEnd(String(endDate)));
+    } else if (layerID === 'VIIRS_ACTIVE_FIRES') {
+      const viirs24H = this._map!.findLayerById('VIIRS_ACTIVE_FIRES');
+      viirs24H.visible = false;
+      const viirs1Y = this._map!.findLayerById('VIIRS1Y') as any;
+      viirs1Y.sublayers.items[0].definitionExpression = defExpression;
+      const viirsIds = VIIRSLayerIDs.map(l => l.id);
+      this._map!.layers.forEach(l => {
+        if (viirsIds.includes(l.id)) {
+          l.visible = l.id === 'VIIRS1Y';
+        }
+      });
+      //sync the new date with redux
+      store.dispatch(setViirsStart(String(startDate)));
+      store.dispatch(setViirsEnd(String(endDate)));
     }
-
-    return;
   }
 
   initializeAndSetVIIRSLayers(): any {
-    const viirsLayers = VIIRSLayerIDs.map(({ layerID, url }) => {
-      return new FeatureLayer({
-        id: layerID,
+    const viirsLayers = VIIRSLayerIDs.map(({ id, url, layerIds }) => {
+      return new MapImageLayer({
+        id: id,
         url,
-        visible: false
+        visible: false,
+        sublayers: [
+          {
+            id: layerIds[0],
+            visible: true
+          }
+        ]
       });
     });
     return viirsLayers;
   }
 
   initializeAndSetMODISLayers(): any {
-    const modisLayers = MODISLayerIDs.map(({ layerID, url }) => {
-      return new FeatureLayer({
-        id: layerID,
+    const modisLayers = MODISLayerIDs.map(({ id, url, layerIds }) => {
+      return new MapImageLayer({
+        id: id,
         url,
-        visible: false
+        visible: false,
+        sublayers: [
+          {
+            id: layerIds[0],
+            visible: true
+          }
+        ]
       });
     });
     return modisLayers;
   }
 
-  setMODISDefinedRange(layer: any, sublayerType: string): void {
-    if (!this._map) {
-      return;
-    }
-
-    const MODIS24 = layer.sublayers.items.filter(
-      (sublayer: Sublayer) => sublayer.title === 'Global Fires (MODIS) 24 hrs'
-    );
+  setMODISDefinedRange(sublayerType: string): void {
+    //Turn off 1Y layer as it does not apply for defined range controls
+    const MODIS1Y = this._map?.findLayerById('MODIS1Y');
+    MODIS1Y!.visible = false;
+    const MODIS24 = this._map!.findLayerById('MODIS_ACTIVE_FIRES');
 
     switch (sublayerType) {
       case '24 hrs':
         {
           MODIS24.visible = true;
-          MODISLayerIDs.forEach(({ layerID }) => {
-            const specificLayer = this._map?.findLayerById(layerID);
+          MODISLayerIDs.forEach(({ id }) => {
+            const fireLayer = this._map?.findLayerById(id);
```

Comment:
It's sort of nitpicky but I was thinking `fireLayer` could be renamed to `modisLayer`

Reply:
ya I like it, I like super specific names so when we read the code we know exactly what is going on